impl/src/main/java/io/jsonwebtoken/impl/lang/Services.java

```diff
@@ -22,12 +22,35 @@
 import java.util.List;
 import java.util.ServiceLoader;
 
+import static io.jsonwebtoken.lang.Collections.arrayToList;
+
 /**
  * Helper class for loading services from the classpath, using a {@link ServiceLoader}. Decouples loading logic for
  * better separation of concerns and testability.
  */
 public final class Services {
 
+    private static final List<ClassLoaderAccessor> CLASS_LOADER_ACCESSORS = arrayToList(new ClassLoaderAccessor[] {
```

Comment:
I think this class would be much cleaner and reduce cyclomatic complexity / loops if we wrapped the list in our own delegating ClassLoader implementation.   Thoughts?

Reply:
I was going to go down that path, but I figured this option was less complex. Mostly because I'm unaware of any gotchas (ClassLoader is a 3000+ abstract class).  I'm guessing the SecurityManager could come into play 🤷‍♂